js/utils.js

```diff
@@ -285,6 +297,180 @@ Fliplet.Registry.set('dynamicListUtils', function() {
     return Promise.resolve(fields);
   }
 
+  function getFiles(data) {
+    var cacheKey = JSON.stringify(data.query);
+
+    if (!cachedFiles[cacheKey]) {
+      cachedFiles[cacheKey] = Fliplet.Media.Folders.get(data.query)
+        .then(function (response) {
+          response.files.forEach(function (file) {
+            if (file.isEncrypted) {
+              file.url = Fliplet.Media.authenticate(file.url);
+            }
+          });
+
+          return response;
+        })
+        .catch(function () {
+          return Promise.resolve({ files: [], folders: [] });
+        });
+    }
+
+    return cachedFiles[cacheKey]
+      .then(function(response) {
+        if (!data.field) {
+          return data.record;
+        }
+
+        var entryData = data.record.data;
+        var column = data.field.column;
+
+        if (isValidImageUrl(entryData[column])) {
+          // Record data doesn't need updating
+          return data.record;
+        }
+
+        var urlEdited = _.some(response.files, function(file) {
+          if (!_.compact(entryData[column]).length) {
+            return;
+          }
+
+          if (entryData[column] && file.name.indexOf(entryData[column]) !== -1) {
+            // File found
+            entryData[column] = file.url;
+            return true;
+          } else if (Static.RegExp.number.test(entryData[column])
+            && parseInt(entryData[column], 10) === file.id) {
+            entryData[column] = file.url;
+            return true;
+          }
+        });
+
+        if (!urlEdited) {
+          entryData[column] = '';
+        }
+
+        return data.record;
+      });
+  }
+
+  function updateRecordFiles(options) {
+    options = options || {};
+
+    var records = options.records || [];
+    var config = options.config || {};
+    var forComments = !!options.forComments;
+
+    if (forComments && !config.userPhotoColumn) {
+      return Promise.resolve(records);
+    }
+
+    var filePromises = [];
+
+    _.forEach(records, function(record) {
+      var defaultData = {
+        query: {},
+        record: record,
+        field: undefined
+      };
+
+      if (!forComments) {
+        _.forEach([config['summary-fields'], config.detailViewOptions], function (fields) {
+          _.forEach(fields, function(field) {
+            if (field.type !== 'image') {
+              return;
+            }
+
+            switch (field.imageField) {
+              case 'app':
```

Comment:
this `case 'app'` and the `case 'organization'` can probably be removed since you removed the options in the dropdown.

Reply:
The app setting currently exists in widget instances, so I think the build output needs to support the behavior.